### PR TITLE
- refactored facelets files using templates and includes

### PR DESCRIPTION
**Pull request by phillipross**

### CODE REVIEW

File: src/main/java/org/pac4j/demo/j2e/ui/ProfileView.java [hunk 1]
```diff
@@ -55,10 +54,41 @@ public List getProfiles() {
     }
 
 
+    /**
+     * Gets the Google OIDC profile if it exists.
+     *
+     * @return the Google OIDC profile
+     */
+    public GoogleOidcProfile getGoogleOidcProfile() {
```
Comment (leleuj) [inline src/main/java/org/pac4j/demo/j2e/ui/ProfileView.java]: I really liked the `ProfileView` as a generic helper to access the profiles, something I can put in j2e-pac4j v3.0. What about a `getProfile(Class)` method? I found these methods too specific...

**Comment by leleuj** [comment]
Good! I'm merging it despite I have one question about the `ProfileView`, we'll answer it in a future PR...

File: src/main/java/org/pac4j/demo/j2e/ui/ProfileView.java (hunk 1 above)
Comment (phillipross) [inline src/main/java/org/pac4j/demo/j2e/ui/ProfileView.java]: Yes, you're correct, and I had intended to do that, but I ran into a few problems with dereferencing in the facelets files.  Either it's not possible to specify a class literal in the JSF expression language, or I haven't found the proper way to do it yet.  A workaround I'm trying is to have a `getProfile(String)` which the facelets files can use to invoke with a String expression of the classname, and then this method can lookup the class and invoke the `getProfile(Class)` if the class is found.  I'll submit another pull request after I make these changes.

Comment (leleuj) [inline src/main/java/org/pac4j/demo/j2e/ui/ProfileView.java]: Excellent! Thanks

**Pull request merged**